api/v1alpha1/fenceagentsremediation_params.go

```diff
@@ -0,0 +1,416 @@
+		return admission.Warnings{}, err
+	}
+
+	// No need to check casting success because it must be either a FenceAgentsRemediationTemplate or a FenceAgentsRemediation
+	metaObj, _ := new.(metav1.Object)
+
+	aggregated := utilErrors.NewAggregate([]error{
+		v.validateAgentName(spec.Agent),
+		v.validateStrategy(spec.RemediationStrategy),
+		v.validateTemplateParameters(spec),
+		v.validateFenceAgentForNodes(ctx, metaObj.GetNamespace(), spec),
+	})
+
+	return admission.Warnings{}, aggregated
+}
+
+func (v *customValidator) getSpec(new runtime.Object) (*FenceAgentsRemediationSpec, error) {
+	switch obj := new.(type) {
+	case *FenceAgentsRemediationTemplate:
+		spec := obj.Spec.Template.Spec
+		return &spec, nil
+	case *FenceAgentsRemediation:
+		spec := obj.Spec
+		return &spec, nil
+	default:
+		return nil, fmt.Errorf("unsupported object type %T", new)
+	}
+}
+
+func (v *customValidator) validateAgentName(agent string) error {
+	exists, err := agentValidator.ValidateAgentName(agent)
+	if err != nil {
+		return utilErrors.NewAggregate([]error{
+			fmt.Errorf("Failed to validate fence agent: %s. You might want to try again.", agent),
+			err,
+		})
+	}
+	if !exists {
+		return fmt.Errorf("unsupported fence agent: %s", agent)
+	}
+	return nil
+}
+
+func (v *customValidator) validateStrategy(farRemStrategy RemediationStrategyType) error {
+	if farRemStrategy == OutOfServiceTaintRemediationStrategy && !isOutOfServiceTaintSupported {
+		return fmt.Errorf("%s remediation strategy is not supported at kubernetes version lower than 1.26, please use a different remediation strategy", OutOfServiceTaintRemediationStrategy)
+	}
+	return nil
+}
+
+// validateTemplateParameters validates template syntax in shared parameters and collects all errors
+func (v *customValidator) validateTemplateParameters(spec *FenceAgentsRemediationSpec) error {
+	var validationErrors []error
+
+	// Validate template syntax in shared parameters
+	for paramName, paramValue := range spec.SharedParameters {
+		if _, err := template.RenderParameterTemplate(paramValue, "dummy-node-name"); err != nil {
+			validationErrors = append(validationErrors, fmt.Errorf("invalid template syntax in shared parameter %s: %w", paramName, err))
+		}
+	}
+
+	return utilErrors.NewAggregate(validationErrors)
+}
+
+// validateFenceAgentForNodes validates fence agent parameters for all the nodes defined in the spec
+// by creating temporary FAR CRs and using BuildFenceAgentParams
+func (v *customValidator) validateFenceAgentForNodes(ctx context.Context, namespace string, spec *FenceAgentsRemediationSpec) error {
+
+	// Check if spec has any parameters at all
+	hasSharedParams := len(spec.SharedParameters) > 0
+	hasNodeParams := len(spec.NodeParameters) > 0
+	hasSecrets := spec.SharedSecretName != nil || spec.NodeSecretNames != nil
+
+	// If template has no parameters or secrets, template is considered invalid
+	if !hasSharedParams && !hasNodeParams && !hasSecrets {
+		err := errors.New(errorMissingParams)
```

Comment:
IIUC validateFenceAgentForNodes will be used for far and farTemplate, so we should differentiate here the logic, since NodeTemplate is only used or relevant for farTemplate.
  between the echecks not be related only to the template (`invalid template: mandatory parameters are missing`)

Reply:
Fair enough, since basically we check the spec here that is shared for both Template and CR I've updated the error to reflect that.